### PR TITLE
Support new crowbar installer

### PR DESCRIPTION
**Pull request by jdsn**
Supporting the new crowbar installer in mkcloud and qa_crowbarsetup.

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
last run succeeded (instonly), will have to make a full run and see if sth breaks

**Comment by jdsn** [comment]
just rebasing on latest master

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1517,33 +1529,20 @@ function do_installcrowbar()
...
 
-    # Make sure install finished correctly
-    if ! [ -e /opt/dell/crowbar_framework/.crowbar-installed-ok ]; then
-        tail -n 90 /root/screenlog.0
-        complain 89 "Crowbar \".crowbar-installed-ok\" marker missing"
-    fi
-
     ensure_packages_installed crowbar-barclamp-tempest
-    # Force restart of crowbar
-    service crowbar stop
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: why do you remove the crowbar restart ? this is going to break cloud 4-5

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1517,33 +1529,20 @@ function do_installcrowbar()
 EOF
         service smb restart
     fi
-    wait_for 300 5 '[ -e /tmp/chef-ready ]' "waiting for chef-ready"
     rpm -Va crowbar\*
 
-    # Make sure install finished correctly
-    if ! [ -e /opt/dell/crowbar_framework/.crowbar-installed-ok ]; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: that test is needed for cloud4+5?

**Comment by jdsn** [comment]
@dirkmueller this branch currently is about cloud6 installer only. Once it works we have to add it as an alternative procedure into qa_crowbarsetup.sh.
Thats why there are the labels wip and do not merge (because it is not ready, we are just experimenting).

**Comment by jdsn** [comment]
Note: this PR will take care that only milestones 1-7 of cloud 6 are installed with the legacy install method. All other cloud 6 sources will switch to the new installer (develcloud as well).

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1483,29 +1483,83 @@ EOF
...
         instcmd="/opt/dell/bin/install-chef-suse.sh $instparams"
     fi
-    do_set_repos_skip_checks
 
     cd /root # we expect the screenlog.0 file here
     echo "Command to install chef: $instcmd"
     intercept "install-chef-suse.sh"
 
     rm -f /tmp/chef-ready
-    rpm -Va crowbar\*
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: why is that being removed?

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -1483,29 +1483,83 @@ EOF
...
 {
     local instparams="$1 --verbose"
-    pre_hook $FUNCNAME
     local instcmd
     if [ -e /tmp/install-chef-suse.sh ]; then
         instcmd="/tmp/install-chef-suse.sh $instparams"
     else
         instcmd="/opt/dell/bin/install-chef-suse.sh $instparams"
     fi
-    do_set_repos_skip_checks
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: why is that being removed?

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -1483,29 +1483,83 @@ EOF
...
+        do_installcrowbar_cloud6plus
+    else
+        do_installcrowbar_legacy $@
+    fi
+    rpm -Va crowbar\*
+
+    ## common code - installer agnostic
+
+    [ -e /etc/profile.d/crowbar.sh ] && . /etc/profile.d/crowbar.sh
+    ensure_packages_installed crowbar-barclamp-tempest
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: that code doesn't make sense in the new function, it should be in _legacy.

File: scripts/qa_crowbarsetup.sh (hunk 3 above)
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: why is that line removed? you're doing it before and after in the new installer as well..

File: scripts/qa_crowbarsetup.sh [hunk 6]
```diff
@@ -1483,29 +1483,83 @@ EOF
...
+
+
+function do_installcrowbar()
+{
+    intercept "crowbar-installation"
+    pre_hook $FUNCNAME
+    do_set_repos_skip_checks
+
+    rpm -Va crowbar\*
+    if iscloudver 6plus && [[ $cloudsource =~ ^M[1-7]$ ]] ; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: that condition doesn't make sense.. I guess you meant 

```bash
if iscloudver 6plus || (iscloudver 6 && ! [[ $cloudsource =~ ^M[1-7]$ ]]) ; then
```

or something similar (just as an idea, not syntactically correct)

**Comment by dirkmueller** [comment]
@jdsn Is this still WIP?

File: scripts/qa_crowbarsetup.sh (hunk 6 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: right, thanks

**Comment by jdsn** [comment]
> @jdsn Is this still WIP?

Yes, @MaximilianMeister is still working on the crowbar-core package, so the topic 'new installer' is WIP.
But this PR can be merged right away once my final M7 check passes.

File: scripts/qa_crowbarsetup.sh (hunk 5 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: this is not the new function but the shared one

Why should `ensure_packages_installed crowbar-barclamp-tempest` not be executed when the new installer is being used?

File: scripts/qa_crowbarsetup.sh (hunk 4 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: its not removed, it is 'moved' to the shared code section, because it is needed for both installation methods

File: scripts/qa_crowbarsetup.sh (hunk 3 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: The point of this is to do the rpm verification before and after the installation.
Thats why it moved to the function that calls the installation (new or legacy). This reduced code duplication because we do not have to call it from both installation functions.

**Comment by rsalevsky** [comment]
+1

**Comment by jdsn** [comment]
Successful manual testruns (incl. updated crowbar packages)
* Milestone 7 (using the legacy installer code path)
* develcloud6 (using the new installer code path)

**Comment by MaximilianMeister** [comment]
:+1:

File: scripts/qa_crowbarsetup.sh (hunk 5 above)
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: Because that package doesn't exist anymore?

File: scripts/qa_crowbarsetup.sh [hunk 7]
```diff
@@ -1483,29 +1483,85 @@ EOF
...
+
+    # Make sure install finished correctly
+    if ! [ -e /opt/dell/crowbar_framework/.crowbar-installed-ok ]; then
+        tail -n 90 /root/screenlog.0
+        complain 89 "Crowbar \".crowbar-installed-ok\" marker missing"
+    fi
+
+    # Force restart of crowbar
+    service crowbar stop
+    service crowbar status || service crowbar start
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: there is ensure_packages_installed crowbar-barclamp-tempest missing here. the reason for the crowbar restart is that a new barclamp got installed. 

Now the restart order is screwed up.

Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: thanks for the info. would be good though if that info had been in the comment above.
